lib/src/http_session.dart

```diff
@@ -61,34 +70,83 @@ class HttpSession implements IOClient {
     Object? body,
     Encoding? encoding,
   }) async {
-    return _updateResponse(await _httpDelegate.delete(
+    Map<String, String> headers0 = {
+      "Cookie": _getCookieHeader(url.host, url.path)
+    };
+    headers0.addAll(headers ?? {});
+    return _updateResponse(_httpDelegate.delete(
       url,
-      headers: _getCookie(headers),
+      headers: headers0,
       body: body,
       encoding: encoding,
     ));
   }
 
+  Future<http.Response> _sendRequest(String method, Uri url, int timeToLive,
```

Comment:
I get a little bit confused when I see 'timeToLive', do you think this is the right word to describe this behavior?

Reply:
Replaced it with `numRecurseLeft`